app/src/commonMain/kotlin/io/github/cgr/app/VirtualMachine.kt

```diff
@@ -16,12 +27,10 @@ class VirtualMachine(val program: Program) {
         }
     }
 
-    fun jump(address: Int) {
-        programCounter = address - 1 // -1 because pc will be incremented at the end of the loop
-    }
-
-    fun jumpRelative(offset: Int) {
-        programCounter += offset - 1 // -1 because pc will be incremented at the end of the loop
+    fun jump(label: Int) {
+        val index = labels[label]
+        if (index == -1) throw IllegalArgumentException("Label $label not found")
+        programCounter = index
```

Comment:
Is the `-1` not necessary anymore?

Reply:
Huh? see the line where the labels array is constructed

Reply:
No, I mean that this used to say `programCounter = index - 1`. Why is that `- 1` not present anymore?

Reply:
because we don't wanna execute the label (its a NOP, but still wastes performance)